src/main/java/spoon/support/util/internal/ElementNameMap.java

```diff
@@ -124,15 +166,21 @@ public boolean containsKey(Object key) {
 	 * @param newKey the new key
 	 */
 	public void updateKey(String oldKey, String newKey) {
-		T type = map.remove(oldKey);
-		if (type != null) {
-			map.put(newKey, type);
+		InsertOrderWrapper<T> wrapper = map.remove(oldKey);
+		if (wrapper != null) {
+			map.put(newKey, wrapper);
 		}
 	}
 
 	@Override
 	public Set<Entry<String, T>> entrySet() {
-		return map.entrySet();
+		return entriesByInsertionOrder().collect(Collectors.toCollection(LinkedHashSet::new));
+	}
+
+	private Stream<Entry<String, T>> entriesByInsertionOrder() {
+		return map.entrySet().stream()
+				.sorted(Comparator.comparing(entry -> entry.getValue().insertionNumber))
+				.map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey(), entry.getValue().value));
```

Comment:
```suggestion
				.map(entry -> Map.entry(entry.getKey(), entry.getValue().value));
```

Unless you need serializability or adjust its value.

Reply:
I didn't know that utility method existed. My brain runs on Java 8.